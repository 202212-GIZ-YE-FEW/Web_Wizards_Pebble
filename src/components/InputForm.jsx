import { Input } from "react-flatifycss";

import "flatifycss/dist/css/flatify-min.css";
import styles from "./InputForm.module.css";
const InputForm = ({
    type,
    placeholder,
    label,
    name,
    hasFloatingLabel,
    togglePasswordLabel,
    togglePassword,
    width,
}) => {
    const InputWidth = width ? `w-${width}` : "w-full";

    return (
        <div className={`px-3 mb-3 ${InputWidth} `}>
            <Input
                autoFocus
                hasFloatingLabel={hasFloatingLabel}
                label={label}
                name={name}
                placeholder={placeholder}
                type={type}
                className={styles.InputForm}
                togglePassword={togglePassword}
                togglePasswordLabel={togglePasswordLabel}
            />
        </div>
    );
};

export default InputForm;
