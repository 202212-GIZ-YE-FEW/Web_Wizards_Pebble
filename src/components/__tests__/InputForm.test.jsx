import renderer from "react-test-renderer";

import InputForm from "../InputForm";

it("renders correctly", () => {
    const tree = renderer.create(<InputForm />).toJSON();
    expect(tree).toMatchSnapshot();
});
