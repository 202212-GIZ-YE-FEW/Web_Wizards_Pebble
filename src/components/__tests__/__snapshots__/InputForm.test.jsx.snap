// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-3 mb-3 w-full "
>
  <div
    className="input__InputWrapper-sc-1hj52gj-0 hyTiyQ input-wrapper"
  >
    <input
      autoFocus={true}
      className="text-input !bg-white !border-2 !border-[black] rounded-lg shadow-none focus:[&:not(#password)]:!border-[3px] focus:[&:not(#password)]:!border-[black] focus:ring-0"
      id="ea014c25-724c-5b"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      pattern={null}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
